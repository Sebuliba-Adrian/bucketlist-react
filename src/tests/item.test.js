import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../components/items/item';


const props = {
  key: 1,
  selectedBucketlist: {
    id: 1,
    title: 'Selected Bucketlist',
    description: 'This is the selected bucketlist',
    created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
  },
  item: {
    id: 1,
    title: 'Test Item',
    description: 'This is a test item',
    created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    status: false,
  },
  request: jest.fn(),
};
it('renders correctly', () => {
  const item = renderer.create(<Item {...props} />).toJSON();
  expect(item).toMatchSnapshot();
});
