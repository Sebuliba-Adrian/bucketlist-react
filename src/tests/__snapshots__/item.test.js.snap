// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="list-group-item list-group-item-action flex-column align-items-start"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-2 hidden-sm-down"
        >
          <div
            className="text-center date-display"
          >
            <div
              className="date-display-month text-white bg-cool-blue"
            >
               Sep
            </div>
            <div
              className="date-display-day"
            >
              <h4>
                 20
              </h4>
            </div>
            <div
              className="date-display-year"
            >
               2017
            </div>
          </div>
        </div>
        <div
          className="col-md-10 item-data"
        >
          <div
            className="d-flex w-100 justify-content-between"
          >
            <p>
              <small
                className="text-muted"
              >
                Pending
              </small>
            </p>
            <div>
              <div
                className="fa fa-lg fa-pencil-square-o mr-2"
                data-backdrop="static"
                data-keyboard="false"
                data-target="#1"
                data-toggle="modal"
              />
              <div
                className="fa fa-lg fa-trash-o ml-2"
                data-backdrop="static"
                data-keyboard="false"
                data-target="#deleteItemModel1"
                data-toggle="modal"
              />
            </div>
          </div>
          <h5
            className="mb-1 text-cool-blue"
          >
            Test Item
          </h5>
          <h6
            className="text-muted"
          >
            This is a test item
          </h6>
          <div>
            <small
              className="text-muted"
            >
              <strong>
                Updated: 
              </strong>
               20 Sep 2017 11:31:50 GMT
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id={1}
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
            Edit a item
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <label
            htmlFor="tiltle"
          >
            Title
          </label>
          <div
            className="input-group"
            onChange={[Function]}
          >
            <input
              className="form-control"
              id="tiltle"
              name="title"
              onChange={[Function]}
              type="text"
              value="Test Item"
            />
          </div>
          <br />
          <label
            htmlFor="description"
          >
            Description
          </label>
          <div
            className="input-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              onChange={[Function]}
              type="text"
              value="This is a test item"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label
              htmlFor="status"
            >
              Status
            </label>
            <select
              className="form-control"
              id="status"
              name="status"
              onChange={[Function]}
              value="0"
            >
              <option
                value="1"
              >
                Finished
              </option>
              <option
                value="0"
              >
                Pending
              </option>
            </select>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="deleteItemModel1"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
            Delete
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          Are you sure you want to delete?
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
